src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int SWITCH_BOUND = 5;
+    private static final int RADIUS_OF_DEFAULT_FIGURE = 10;
+    private static final int ORIGIN = 1;
+    private static final int MAX_VALUE = 100;
+    private static final String COLOR_OF_DEFAULT_FIGURE = "white";
+
+    public Figure getRandomFigure() {
+
+        ColorSupplier colorSupplier = new ColorSupplier();
+
+        int index = new Random().nextInt(SWITCH_BOUND);
+        String color = colorSupplier.getRandomColor();
+
+        switch (index) {
+            case 0 -> {
+                return new Circle(randomNumber(),
+                        color);
+            }
+            case 1 -> {
+                return new IsoscelesTrapezoid(randomNumber(),
+                        randomNumber(),
+                        randomNumber(),
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example, create variables for the random numbers and then pass these variables to the constructor.